artview/components/nav.py

```diff
@@ -258,40 +266,91 @@ def AdvanceFileSelect(self, findex):
 
     def goto_prev_file(self):
         self.fileindex = self.fileindex - 1
-        self.act_prev.setToolTip(
-            'Previous file: %s'%os.path.basename(self.Vfilelist.value[self.fileindex - 1]))
-        self.act_next.setToolTip(
-            'Next file: %s'%os.path.basename(self.Vfilelist.value[self.fileindex + 1]))
         self.AdvanceFileSelect(self.fileindex)
+        self._update_tools()
 
     def goto_next_file(self):
         self.fileindex = self.fileindex + 1
-        self.act_prev.setToolTip(
-            'Previous file: %s'%os.path.basename(self.Vfilelist.value[self.fileindex - 1]))
-        self.act_next.setToolTip(
-            'Next file: %s'%os.path.basename(self.Vfilelist.value[self.fileindex + 1]))
         self.AdvanceFileSelect(self.fileindex)
+        self._update_tools()
 
-    def NewFileList(self, variable, value, strong):
-        '''respond to change in radar.'''
-        if self.Vfilelist.value is None:
+    def _openfile(self, filename=None):
+        '''Open a file via a file selection window.'''
+        print("Opening file " + filename)
+
+        # Read the data from file
+        radar_warning = False
+        grid_warning = False
+
+        try:
+            radar = pyart.io.read(filename, delay_field_loading=True)
+            # Add the filename for Display
+            radar.filename = filename
+            self.Vradar.change(radar)
+            return
+        except:
+            try:
+                radar = pyart.io.read(filename)
+                # Add the filename for Display
+                radar.filename = filename
+                self.Vradar.change(radar)
+                return
+            except:
+                import traceback
+                print(traceback.format_exc())
+                radar_warning = True
+
+        try:
+            grid = pyart.io.read_grid(
+                filename, delay_field_loading=True)
+            self.Vgrid.change(grid)
             return
+        except:
+            try:
+                grid = pyart.io.read_grid(filename)
+                self.Vgrid.change(grid)
+                return
+            except:
+                import traceback
+                print(traceback.format_exc())
+                grid_warning = True
+
+        if grid_warning or radar_warning:
+            msg = "Py-ART didn't recognize this file!"
+            common.ShowWarning(msg)
+        else:
+            msg = "Could not open file, invalid mode!"
+            common.ShowWarning(msg)
+        return
 
+    def NewFilelist(self, variable, value, strong):
+        '''respond to change in filelist.'''
         if strong:
-            self.Vfilelist.change(value)
             self._update_tools()
 
+    def NewFile(self, variable, value, strong):
+        '''Respond to change in a container (radar or grid).'''
+        if hasattr(variable.value, 'filename'):
+            # Update the info label.'''
+            self.filename = variable.value.filename
```

Comment:
By just calling ```_update_tools``` the ```self.fileindex``` should be updated, no?

I wonder if we took ```_update_tools``` out of all ```goto_*``` functions and just put it in ```_openfile```?